### DISCUSSION TITLE
Unpackaged, self-contained apps share WinAppSDK dlls?

### DISCUSSION THREAD

**Discussion by stefansjfw** [microsoft/WindowsAppSDK#2936] [Q&A] [answered]
Hi,

Scenario is following: I have 2 (or more) unpackaged, self-contained WinAppSDK apps. In order to save disk space for apps users, I want to ship WinAppSDK DLLs only once (let's say in `<app_install_path>/dll/` dir). Is there a way to specify to apps to look for DLLs at that specific place? I tried adding that path to PATH env var, doesn't work. I tried creating symbolic links in apps dirs pointing to dll/ DLLs, that works, but I'd prefer to have something like LD_LIBRARY_PATH env var in Linux. Having both apps in the same wouldn't work for me.

Thanks in advance

**Comment by DarranRowe** [comment, marked as answer]
To be honest, symbolic links or hard links are probably the better option here.
Because the Windows App SDK components are WinRT components, they need to be properly registered for the application to be able to use them. This normally occurs when you load the Windows App SDK package (either through the bootstrap API or through your application being packaged).
However, when your application is self contained, application needs to use another means of registering the Windows App SDK components. This is where registration free WinRT comes in. The components use the application manifest file to register the component.
So if you look at your application manifest, you will find a bunch of entries that list the file name and the activatable classes.
There are two main issues here. First, I'm not sure if the entries allow relative paths. Second, the project system embeds the manifest file into the executable, so editing it is a pain.

**Reply by stefansjfw** [reply]
thanks for the answer. Hard links are the best option to go for after some research
